test/unit/font_substitution_spec.js

```diff
@@ -0,0 +1,356 @@
+/* Copyright 2022 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createIdFactory } from "./test_utils.js";
+import { getFontSubstitution } from "../../src/core/font_substitutions.js";
+
+describe("getFontSubstitution", function () {
+  const cache = {
+    has: () => false,
+    set: () => {},
+    get: () => undefined,
+  };
+  const idFactory = createIdFactory(0);
+  const localFontPath = "/tmp/";
+
+  it("should substitute an unknow font", () => {
+    const fontName = "Foo";
+    const fontSubstitution = getFontSubstitution(
+      cache,
```

Comment:
Here, and elsewhere below, can we please use the *correct* type for the cache to prevent any future issues/inconsistencies?
```suggestion
      new Map(),
```